src/main/java/com/puppycrawl/tools/checkstyle/utils/ScopeUtil.java

```diff
@@ -266,7 +266,7 @@ public static boolean isLocalVariableDef(DetailAST node) {
         }
 
         if (node.getType() == TokenTypes.RESOURCE) {
-            localVariableDef = true;
+            localVariableDef = node.getChildCount() != 1;
```

Comment:
Should it be `node.getChildCount() > 1` ?

Reply:
Good catch, fixed